src/main/groovy/com/github/jengelman/gradle/plugins/shadow/ShadowApplicationPlugin.groovy

```diff
@@ -93,8 +93,9 @@ class ShadowApplicationPlugin implements Plugin<Project> {
     }
 
     protected void configureInstallTask(Project project) {
-        project.tasks.named(SHADOW_INSTALL_TASK_NAME).configure { installTask ->
-            installTask.doFirst { Sync task ->
+        project.tasks.named(SHADOW_INSTALL_TASK_NAME, Sync).configure { task ->
```

Comment:
Not sure this is the proper way, I just see
```java
@DisableCachingByDefault(because = "Not worth caching")
public abstract class Sync extends AbstractCopyTask 
```

Reply:
If the only reason for this `Sync` task to me marked as not compatible with the configuration cache is the usage of `project.ant.chmod` in its `doLast {}`  action, then this can be replaced by configuring the file mode using:

```groovy
task.eachFile {
    if (it.path == "bin/${javaApplication.applicationName}") {
        mode = 0x755
    }
}
```

Reply:
Seems can't, see https://github.com/Goooler/shadow/commit/2a91f58204ca49b720bdf2556026f71deca49b79

Reply:
The `eachFile {}` configuration should not be in the `doLast {}` action but in the task configuration.

Reply:
Same result in https://github.com/Goooler/shadow/commit/c7e3953a219a5b198f51f8563b74fe09d74c0aef

Reply:
Then something else must hold a reference to the project.
Probably `javaApplication`?

What about when you do this?
```groovy
def appName = javaApplication.applicationName
 task.eachFile {
  if (it.path == "bin/${appName}") {
    it.mode = 0x755
  }
}
```

Reply:
Tryied in https://github.com/Goooler/shadow/pull/16

Reply:
That won't work because the `AntBuilder` is not supported by CC.
You need to combine both capturing the app name in a provider and use `eachFile {}` instead of `ant.chmod`

Reply:
It works now in 3543ca6559aa81d33ed1f4d7338d150a2da3fd7c.

Reply:
Thanks!